include/lbann/utils/random.hpp

```diff
@@ -59,6 +59,21 @@ fast_rng_gen& get_fast_generator();
  * @note If compiling with OpenMP, this is stored in a threadprivate variable.
  */
 rng_gen& get_data_seq_generator();
+
+/**
+ * Return a reference to the global LBANN random number generator used
+ * for shuffling the data samples within each mini-batch
+ * @note This is stored in a threadprivate variable.
```

Comment:
`thread_local`, not threadprivate.

Should be clear that this is not meant to be used from OpenMP threads.